datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -49,7 +50,7 @@ macro_rules! binary_bitwise_array_op {
 /// like int64, int32.
 /// It is used to do bitwise operation on an array with a scalar.
 macro_rules! binary_bitwise_array_scalar {
```

Comment:
I've changed `binary_bitwise_array_scalar` to pass function as `expr` to do casting, because `wrapping_shr` and `wrapping_shl` requires u32. I tried to do it with `            let right: $TYPE = scalar.try_into().unwrap();` but I found that it will show a error because it's not possible to cast all numeric types to `u32`.

Reply:
I think it makes sense -- thank you